bridge/client/app/_components/ktb-project-settings/ktb-project-settings.component.ts

```diff
@@ -212,82 +194,43 @@ export class KtbProjectSettingsComponent implements OnInit, OnDestroy, PendingCh
-              NotificationType.SUCCESS,
-              'The Git upstream was changed successfully.'
-            );
-
-            this.pendingChangesSubject.next(true);
-            this.isProjectFormTouched = false;
-          },
-          (err) => {
-            this.isGitUpstreamInProgress = false;
-            this.notificationsService.addNotification(
-              NotificationType.ERROR,
-              `<div class="long-note align-left p-3">The Git upstream could not be changed:<br/><span class="small">${err.error}</span></div>`
-            );
-          }
-        );
+  public async createProject(): Promise<void> {
+    if (!this.shipyardFile || this.gitDataExtended === undefined) {
+      return;
     }
-  }
+    const fileContent = await FormUtils.readFileContent(this.shipyardFile);
+    if (!fileContent) {
+      return;
+    }
+    this.isCreatingProjectInProgress = true;
 
-  public async createProject(): Promise<void> {
-    if (this.shipyardFile) {
-      this.isCreatingProjectInProgress = true;
-      const fileContent = await FormUtils.readFileContent(this.shipyardFile);
-      if (fileContent) {
-        const shipyardBase64 = btoa(fileContent);
-        const projectName = this.projectNameControl.value;
-        const create$ =
-          this.resourceServiceEnabled && this.gitDataExtended
-            ? this.dataService.createProjectExtended(projectName, shipyardBase64, this.gitDataExtended)
-            : this.dataService.createProject(
-                projectName,
-                shipyardBase64,
-                this.gitData.gitRemoteURL,
-                this.gitData.gitToken,
-                this.gitData.gitUser
-              );
-
-        create$.subscribe(
-          () => {
-            this.projectName = projectName;
-            this.dataService.loadProjects().subscribe(() => {
-              this.isCreatingProjectInProgress = false;
-              this.isProjectFormTouched = false;
-
-              this.router.navigate(['/', 'project', this.projectName, 'settings', 'project'], {
-                queryParams: { created: true },
-              });
-            });
-          },
-          (err) => {
-            const service = this.resourceServiceEnabled ? 'resource-service' : 'configuration-service';
-            const errorMessage = err.error || `please, check the logs of ${service}`;
-            this.notificationsService.addNotification(
-              NotificationType.ERROR,
-              `The project could not be created: ${errorMessage}.`
-            );
-            this.isCreatingProjectInProgress = false;
-          }
+    const shipyardBase64 = btoa(fileContent);
+    const projectName = this.projectNameControl.value;
+
+    this.dataService.createProjectExtended(projectName, shipyardBase64, this.gitDataExtended ?? undefined).subscribe(
+      () => {
+        this.projectName = projectName;
+        this.dataService.loadProjects().subscribe(() => {
```

Comment:
Can we refactor the the nested subscription to to a RXJS pipe and then subscribe once to it?

Reply:
done